toolkit/tools/imagegen/configuration/systemconfig.go

```diff
@@ -66,6 +67,34 @@ func (s *SystemConfig) GetMountpointPartitionSetting(mountPoint string) (partiti
 	return FindMountpointPartitionSetting(s.PartitionSettings, mountPoint)
 }
 
+func (s *SystemConfig) validateUsersAndGroups() (err error) {
+	for _, user := range s.Users {
+		groupMatchFunc := func(groupName interface{}, groupObj interface{}) bool {
+			return groupName == groupObj.(Group).Name
+		}
```

Comment:
Is there a need to define `groupMatchFunc` for each iteration of the loop?
```suggestion
	groupMatchFunc := func(groupName interface{}, groupObj interface{}) bool {
		return groupName == groupObj.(Group).Name
	}

	for _, user := range s.Users {
```

Reply:
Readability is better to me, and this is such a small loop the perf uplift didn't seem important. I originally had it as an inline function in the call to sliceutils but it was a bit hard to read.